cli/flox-rust-sdk/src/models/environment/path_environment.rs

```diff
@@ -240,8 +241,33 @@ impl Environment for PathEnvironment {
     }
 
     fn activation_path(&mut self, flox: &Flox) -> Result<PathBuf, EnvironmentError2> {
-        self.build(flox)?;
-        self.out_link(&flox.system)
+        let manifest_modified_at = 'time: {
```

Comment:
nit: could factor out a compare mtimes function which would also be good for unit testing

Reply:
factored out the "get mtime" part with the defaults here.
Also added a `PathEnvironment::needs_rebuild` which is doing the comparison -- and can be used in unit tests.